index.js

```diff
@@ -126,8 +126,8 @@ RedirectableRequest.prototype.end = function (data, encoding, callback) {
     var currentRequest = this._currentRequest;
     this.write(data, encoding, function () {
       self._ended = true;
-      currentRequest.end(null, null, callback);
     });
+    currentRequest.end(null, null, callback);
     this._ending = true;
```

Comment:
Perhaps swap those two lines actually?

Reply:
Yes, definitely need to be swapped, or we can write after end in the `callback` (let's add a test).

Reply:
You're talking about swapping `_ending` and `_ended` right?

```js
    this.write(data, encoding, function () {
      this._ending = true;
    });
    self._ended = true;
    currentRequest.end(null, null, callback);
```

This makes sense. I think I've overlooked that.

Reply:
@RubenVerborgh, I've swapped the order of ending/ended state.

Reply:
> or we can write after end in the callback (let's add a test).

Not sure what you mean here. We can't call `write` in the callback of `end`, if that's what you're referring to. That'd throw an exception (cannot write, already ended).